functions/src/common/datastore.ts

```diff
@@ -61,10 +61,13 @@ export class Datastore {
    * These attributes are merged with other existing ones if already present.
    */
   async mergeUserProfile(user: functions.auth.UserRecord) {
-    const { uid, email, displayName, photoURL } = user;
+    const {uid, email, displayName, photoURL} = user;
     await this.db_
       .doc(`users/${uid}`)
-      .set({ email, displayName, photoURL }, { merge: true });
+      .set(
+        {email, displayName, photoURL: photoURL?.replace(/=s..-[c]$/g, '')},
```

Comment:
Please move `photoURL?.replace(/=s..-[c]$/g, '')` into its own function (e.g., `trimPhotoResizeSuffix`) and add a note explaining what the regex will trim.

Also, I just tested and the `-c` part isn't required, and the size doesn't necessarily need to be two digits. Since we're not sure of the exact syntax, perhaps we can trim more aggressively with `/=s.*$/`? The preceding ID appears to be alphanumeric only, so I think we're probably ok to assume "=" won't appear in the main part of the URL.

Reply:
added a static function named trimPhotoURLSizeSuffix with the following comment:

`  /**
   * Removes a possible size suffix from the user's photo URL.
   *
   * @param photoURL The user's photo URL.
   */`